### ISSUE TITLE
Unique instance support is flaky

### ISSUE THREAD

**Issue by aidanholm** [luakit/luakit#331]
Luakit does not always receive URIs to load from a secondary instance. If luakit is already running, then `for n in 1 2 3 4; do luakit $n; done` does not always open URIS 1, 2, 3, and 4 in the primary instance.

OS: Crux
GTK: 3.22.16
WebKit: 2.17.4

**Comment by aidanholm** [comment]
User report:
```
https://wiki.gnome.org/Projects/GLib/GApplication/DBusAPI

dbus-monitor path=/org/luakit sender=org.luakit destination=org.luakit

for n in 1 2 3 4; do luakit "$n"; done

when it works:
method call time=1502932286.653814 sender=:1.37 -> destination=org.luakit serial=5 path=/org/luakit; interface=org.gtk.Actions; member=Activate
   string "message"
   array [
      variant          string "open-uri-set {{
[1]="2",
},
}"
   ]
   array [
      dict entry(
         string "cwd"
         variant             array of bytes "/tmp" + \0
      )
   ]


when it fails:
method call time=1502932288.941365 sender=:1.39 -> destination=org.luakit serial=4 path=/org/luakit; interface=org.gtk.Actions; member=DescribeAll
method return time=1502932288.942336 sender=:1.29 -> destination=:1.39 serial=25 reply_serial=4
   array [
      dict entry(
         string "message"
         struct {
            boolean true
            signature "s"
            array [
            ]
         }
      )
   ]
```

**Comment by aidanholm** [comment]
Issue seemed to be caused by a race condition in glib: fixed by patching glib 2.55.1 to use synchronous dbus activate calls.
